### PR TITLE
Add MAppBar

### LINKED ISSUE #13: Add AppBar
- Material Design Guideline
  - [App bars : bottom](https://material.io/design/components/app-bars-bottom.html)
  - [App bars : top](https://material.io/design/components/app-bars-top.html)

### PR DESCRIPTION
**Pull request by shyunMin**
### Description of Change ###
This PR is for adding MTopAppBar and MBottomAppBar

### Issues Resolved ### 
- fixes #13

### API Changes ###
Added
 - MTopAppBar
    - NavigationItem
    - Title
    - TitleColor
    - ActionItems
    Up to 2 item can be shown on app bar, the other items are added into the popup which will show when overflow button is clicked.
top

    - BackgroundImage
    - Prominent
    Prominent app bar is suggested when a user set background image on app bar, and the longer title is supported (maximum 2 lines)
top_1
top_2

    - Theming 
    - Contextual Action Bar
    It can be changed to contextual action bar as follow
      - The bar color changes
      - Navigation icon is replaced with a close icon
      - Top app bar title text converts to contextual action bar text
      - Top app bar actions are replaced with contextual actions
    - Scrolling (Not Supported)

 - MBottomAppBar
    - NavigationItem
    - FloatingActionButtom 
    Users should create MComformant on their app for floating button)
    - BottomAppBarLayoutOption
      - CenteredFAB
      (NavigationItem / FloatingActionButton / 1~2 ActionItems)
      Use bottom app bars on home screens that feature a navigation menu control and a prominent action (such as a FAB). A minimum of one and a maximum of two additional actions can be placed on the opposite side of the bar.
center

      - EndFAB
      (FloatingActionButton / 3~4 ActionItems)
      Use a FAB on secondary screens that require a floating action button and three to four additional actions.
end

      - NoFAB
      (NavigationItem / ~4 ActionItems)
      When no floating action button is needed, the bottom app bar can accommodate a navigation menu icon and up to four actions aligned on the opposing edge.
no

    - Overlapping
      - FAB: Supported 
      - Inset: Not Supported
    - Theming
    - Scrolling (Not Supported)

### Platforms Affected ### 
- Tizen

### Behavioral/Visual Changes ###
None

### Before/After Screenshots ### 
 - MTopAppBar
tab

 - MBottomAppBar
bab

### Tip ###
- MBottomAppBar with virtual keypad
In the material design guide, the bottom app bar can be covered by keyboards and other temporary surfaces. 
To implement this, you can use KeypadOn/Off event in MComformant or you can use SmartEvent providing in Comformant like below.

```
var _conformant = new MConformant(window);
_conformant.KeyPadOn += (s, e) =>
{
     box.UnPack(appbar);
};

_conformant.KeyPadOff += (s, e) =>
{
     box.PackEnd(appbar);
};
```

bab_key

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: src/Tizen.NET.MaterialComponents/Components/MActionItem.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace Tizen.NET.MaterialComponents
+{
+    public class MActionItem
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MActionItem.cs]: Is there no method and event to activate the given Action? e.g. `Active()` and `Clicked`

File: src/Tizen.NET.MaterialComponents/Components/MAppBar.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+            _naviButton = new MButton(this);
+
+            _items.CollectionChanged += OnItemsCollectionChanged;
+            SetLayoutCallback(() => { UpdateChildrenGeometry(); });
+
+            MColors.AddColorSchemeComponent(this);
+        }
+
+        public IList<MActionItem> ActionItems
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MAppBar.cs]:
```suggestion
        public IList<MActionItem> ActionItems => _items;
```

File: src/Tizen.NET.MaterialComponents/Components/MTopAppBar.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                TextStyle = $"DEFAULT = 'font_size={DefaultValues.AppBar.FontSize}'"
+            };
+
+            VisibleItemCount = 2;
+            OverflowPopupToDown = true;
+
+            OnColorSchemeChanged(true);
+        }
+
+        public string Title {
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MTopAppBar.cs]: spacing

**Review by rookiejava — COMMENTED** [review]
It wou;d be more helpful to review if you tell us a little bit more about the difference between the functional/design perspective/what else that you intended.

File: src/Tizen.NET.MaterialComponents/Components/MActionItem.cs (hunk 1 above)
Comment (shyunMin) [reply src/Tizen.NET.MaterialComponents/Components/MActionItem.cs]: Fixed.

File: src/Tizen.NET.MaterialComponents/Components/MAppBar.cs (hunk 2 above)
Comment (shyunMin) [reply src/Tizen.NET.MaterialComponents/Components/MAppBar.cs]: Fixed.

File: src/Tizen.NET.MaterialComponents/Components/MTopAppBar.cs (hunk 3 above)
Comment (shyunMin) [reply src/Tizen.NET.MaterialComponents/Components/MTopAppBar.cs]: Fixed.

**Review by rookiejava — APPROVED** [review]

**Pull request merged**